server/src/test/java/org/candlepin/resource/ConsumerResourceIntegrationTest.java

```diff
@@ -686,4 +690,87 @@ public void testContentAccessExpireRegen() {
+
+        Date beforeUpdateTimestamp = consumerCurator.findByUuid(consumer.getUuid())
+            .getRHCloudProfileModified();
+
+        Product prod = TestUtil.createProduct("Product One");
+        ConsumerInstalledProductDTO updatedInstalledProduct =
+            new ConsumerInstalledProductDTO(prod.getId(), prod.getName());
+        consumer.addInstalledProduct(updatedInstalledProduct);
+        consumerResource.updateConsumer(consumer.getUuid(), consumer, principal);
+
+        Date afterUpdateTimestamp = consumerCurator.findByUuid(consumer.getUuid())
+            .getRHCloudProfileModified();
+
+        assertNotNull(afterUpdateTimestamp);
+        assertNotEquals(beforeUpdateTimestamp, afterUpdateTimestamp);
+    }
+
+    @Test
+    public void testCloudProfileUpdatedOnHypervisorUpdate() {
+        ConsumerDTO consumer = createConsumerDTO("random-consumer", null, null, standardSystemTypeDTO);
+        consumer = consumerResource.create(consumer, principal, null, null, null, true);
+
+        Date beforeUpdateTimestamp = consumerCurator.findByUuid(consumer.getUuid())
+            .getRHCloudProfileModified();
+
+        HypervisorIdDTO updatedHypervisorId = new HypervisorIdDTO();
+        consumer.setHypervisorId(updatedHypervisorId);
+        consumerResource.updateConsumer(consumer.getUuid(), consumer, principal);
+
+        Date afterUpdateTimestamp = consumerCurator.findByUuid(consumer.getUuid())
+            .getRHCloudProfileModified();
+
+        assertNotNull(afterUpdateTimestamp);
+        assertNotEquals(beforeUpdateTimestamp, afterUpdateTimestamp);
+    }
+
+    @Test
+    public void testCloudProfileUpdatedOnSpecificConsumerFactsUpdate() {
+        ConsumerDTO consumer = createConsumerDTO("random-consumer", null, null, standardSystemTypeDTO);
+        consumer = consumerResource.create(consumer, principal, null, null, null, true);
+        Date modifiedDateOnCreate = consumerCurator.get(consumer.getId()).getRHCloudProfileModified();
+
+        ConsumerDTO updatedConsumerDTO = new ConsumerDTO();
+        updatedConsumerDTO.setFact("FACT", "FACT_VALUE");
+        consumerResource.updateConsumer(consumer.getUuid(), updatedConsumerDTO, principal);
+        Date modifiedTSOnUnnecessaryFactUpdate = consumerCurator.get(consumer.getId())
+            .getRHCloudProfileModified();
+
+        assertEquals(modifiedDateOnCreate, modifiedTSOnUnnecessaryFactUpdate);
+
+        updatedConsumerDTO = new ConsumerDTO();
+        updatedConsumerDTO.setFact(CloudProfileFacts.CPU_CORES_PERSOCKET.getFact(), "1");
+        consumerResource.updateConsumer(consumer.getUuid(), updatedConsumerDTO, principal);
+        Date modifiedTSOnNecessaryFactUpdate = consumerCurator.get(consumer.getId())
+            .getRHCloudProfileModified();
+
+        assertNotNull(consumerCurator.get(consumer.getId()).getRHCloudProfileModified());
+        assertNotEquals(modifiedTSOnUnnecessaryFactUpdate, modifiedTSOnNecessaryFactUpdate);
+    }
```

Comment:
It would probably be good to include tests in this class for at least a couple of the fields that are not supposed to update the timestamp, and verify that the timestamp stays the same after the update. Also, same for a couple of facts that are not cloud-profile related

Reply:
Added.